File: app/src/main/java/com/chuongvd/app/signal/data/source/local/datasource/IUserLocalDataSource.java
package com.chuongvd.app.signal.data.source.local.datasource;

import android.arch.lifecycle.LiveData;
import com.chuongvd.app.signal.data.entity.UserEntity;

public interface IUserLocalDataSource {
    void saveUser(UserEntity user);

    LiveData<UserEntity> getUser();

    String getAccessToken();

}

File: app/src/main/java/com/chuongvd/app/signal/data/source/local/datasource/IUserLocalDataSource.kt
package com.chuongvd.app.signal.data.source.local.datasource

import android.arch.lifecycle.LiveData
import com.chuongvd.app.signal.data.entity.UserEntity

interface IUserLocalDataSource {

  val user: LiveData<UserEntity>

  val accessToken: String?
  fun saveUser(user: UserEntity)

}
